subprojects/core/src/main/java/org/gradle/api/artifacts/transform/internal/ArtifactTransforms.java

```diff
@@ -82,26 +84,47 @@ public DependencyTransformRegistration(String from, String to, Class<? extends A
         public Transformer<File, File> getTransformer() {
             ArtifactTransform artifactTransform = DirectInstantiator.INSTANCE.newInstance(type);
             config.execute(artifactTransform);
-            return new DependencyTransformTransformer(artifactTransform, outputProperty);
+            return new DependencyTransformTransformer(artifactTransform, outputProperty, to);
         }
     }
 
     private static class DependencyTransformTransformer implements Transformer<File, File> {
+        private static final Logger LOGGER = Logging.getLogger(DependencyTransformTransformer.class);
+
         private final ArtifactTransform artifactTransform;
         private final JavaMethod<? super ArtifactTransform, File> outputProperty;
+        private final String outputFormat;
 
-        private DependencyTransformTransformer(ArtifactTransform artifactTransform, JavaMethod<? super ArtifactTransform, File> outputProperty) {
+        private DependencyTransformTransformer(ArtifactTransform artifactTransform, JavaMethod<? super ArtifactTransform, File> outputProperty, String outputFormat) {
             this.artifactTransform = artifactTransform;
             this.outputProperty = outputProperty;
+            this.outputFormat = outputFormat;
         }
 
         @Override
         public File transform(File file) {
             if (artifactTransform.getOutputDirectory() != null) {
                 artifactTransform.getOutputDirectory().mkdirs();
             }
-            artifactTransform.transform(file);
-            return outputProperty.invoke(artifactTransform);
+            File output = null;
+            try {
+                artifactTransform.transform(file);
+                output = outputProperty.invoke(artifactTransform);
+                if (output == null) {
+                    reportError(file, "no output file created", null);
+                } else if (!output.exists()) {
+                    reportError(file, "expected output file '" + output.getPath() + "' was not created", null);
+                }
+            } catch (Exception e) {
+                reportError(file, e.getMessage(), e);
+            }
+            return output;
+        }
+
+        private void reportError(File input, String message, Exception e) {
+            LOGGER.error(String.format("Error while transforming '%s' to format '%s' using '%s'%s",
```

Comment:
It shouldn't be the responsibility of this class to report the error, it should just wrap the failure in an exception that adds this context and throw the exception. Something else takes care of dealing with the failure and reporting it.

It's also important that user code sees some kind of exception thrown out of the query methods of `Configuration` when a transformation fails, otherwise it's going to look like the resolution was successful and the result is correct and complete, which of course it is not when a transformation fails.